### PR TITLE
Import into eclipse workspace

### PR DESCRIPTION
**Pull request by ebamberg**
## Description ##

djl.ai did not compile after importing the gradle project into eclipse.
reasons are 2 issues which per default are flagged by eclipse as errors cause this can be seen as bad smells/bad coding practise. 

1) cycle dependencies in the compile-classpath. 
2) package-info.java files exists in src/main/java and in src/test/java in the same package. same file getting compiled to same package in the classpath is a bad smell and get flagged.

Both issue can be configured in eclipse workspace so that eclipse ignore this, but we have to do this after every new import - for example when clean checkout a new version of djl (like version 0.9.0 yesterday)

this can get automated through gradle -> this change set the settings for eclipse workspace and can be execute via "gradlew eclipse" before importing

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,9 @@ libs
 *.dylib
 *.dll
 *.class
+
+# Eclipse
+.settings
+.project
+.classpath
+bin/
```
Comment (stu1130) [inline .gitignore:22]: add a new line here

File: README.md [hunk 2]
```diff
@@ -121,13 +121,33 @@ gradlew build -x test
...
+**Note:** please set your workspace text encoding setting to UTF-8
+
 ## Community
 
 Join our [<img src='https://cdn3.iconfinder.com/data/icons/social-media-2169/24/social_media_social_media_logo_slack-512.png' width='20px' /> slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with the development team, for questions and discussions.
 
 Follow our [<img src='https://cdn2.iconfinder.com/data/icons/social-media-2285/512/1_Twitter_colored_svg-512.png' width='20px' /> twitter](https://twitter.com/deepjavalibrary) to see updates about new content, features, and releases.
 
-关注我们 [<img src='https://www.iconfinder.com/icons/5060515/download/svg/512' width='20px' /> 知乎专栏](https://zhuanlan.zhihu.com/c_1255493231133417472) 获取DJL最新的内容！
+å…³æ³¨æˆ‘ä»¬ [<img src='https://www.iconfinder.com/icons/5060515/download/svg/512' width='20px' /> çŸ¥ä¹Žä¸“æ �](https://zhuanlan.zhihu.com/c_1255493231133417472) èŽ·å�–DJLæœ€æ–°çš„å†…å®¹ï¼�
```
Comment (frankfliu) [inline README.md:150]: It looks like those Chinese characters is trashed due to charset issue.
Please make sure your editor support UTF-8.
```suggestion
关注我们 [<img src='https://www.iconfinder.com/icons/5060515/download/svg/512' width='20px' /> 知乎专栏](https://zhuanlan.zhihu.com/c_1255493231133417472) 获取DJL最新的内容！
```

Comment (stu1130) [inline README.md:150]: revert the change here

File: build.gradle [hunk 3]
```diff
@@ -42,6 +49,22 @@ configure(javaProjects()) {
     if (JavaVersion.current() != JavaVersion.VERSION_1_8) {
         compileJava.options.compilerArgs.addAll(["--release", "8"])
     }
+    
+	apply plugin: 'eclipse'
+    
+	eclipse {
```
Comment (stu1130) [inline build.gradle:55]: maybe indent with 4 spaces here

**Review by stu1130 — COMMENTED** [review]
Thanks for your contribution!

**Comment by frankfliu** [comment]
> 2. package-info.java files exists in src/main/java and in src/test/java in the same package. same file getting compiled to same package in the classpath is a bad smell and get flagged.

We enabled checkstyle, if we remove those package-info.java, checkstyle will fail. I didn't find a way to workaround this issue.

**Comment by ebamberg** [comment]
> > 1. package-info.java files exists in src/main/java and in src/test/java in the same package. same file getting compiled to same package in the classpath is a bad smell and get flagged.
> 
> We enabled checkstyle, if we remove those package-info.java, checkstyle will fail. I didn't find a way to workaround this issue.

this lines in the gradle, will advise eclipse to ignore the package-info files in src/test during building, so its fine for eclipse users

		sourceSets.test.java {
				srcDirs = ["src/test/java"]
				exclude "**/package-info.java"
			}

**Review by frankfliu — APPROVED** [review]
